lib/src/location_layer.dart

```diff
@@ -126,15 +126,9 @@ class _LocationLayerState extends State<LocationLayer>
                   options: MarkerLayerOptions(markers: <Marker>[marker]));
             }),
         widget.options.buttonBuilder(context, _serviceStatus, () async {
-          if (_serviceStatus?.value == LocationServiceStatus.disabled) {
-            if (!await _location.requestService()) {
```

Comment:
Here is an regression. Before the change a dialog was shown to the user if the service was disabled. With the change simply nothing happens.

There is no explicit call in geolocator. Instead https://github.com/Baseflow/flutter-geolocator/issues/254#issuecomment-489051318

That fails because `_initOnLocationUpdateSubscription` early returns if the location service is off. If we remove the check there it works. I'll prepare a MR for that.

Reply:
See #56